### PR TITLE
Balloon alerts - Text for minor actions w/o other feedback

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request

https://user-images.githubusercontent.com/35135081/118361344-e08f8000-b53f-11eb-8da2-d1d6fa95e164.mp4

https://user-images.githubusercontent.com/35135081/118361257-652dce80-b53f-11eb-8fdd-261fccc0ce21.mp4

(The text going underneath the potted plant was fixed)

Currently only applies to beakers changing amounts as a proof of concept of an action that we can't really provide any non-text feedback on, as opposed to something you could have different sounds for.

Inspired by SS14.

## Why It's Good For The Game
In the right place, balloon alerts will let you keep your eyes on the game itself, rather than constantly looking at the chat. This is especially important when chat messages will collapse into each other. This is in most cases warranted, but can make other cases (such as changing beaker amounts) confusing.

## Changelog
:cl:
qol: Changing a reagent container's transfer amount will now give you feedback in the form of text on the container itself.
/:cl:

### CODE REVIEW

File: code/modules/balloon_alert/balloon_alert.dm [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+	animate(
+		alpha = 255,
+		time = BALLOON_TEXT_SPAWN_TIME,
+		easing = CUBIC_EASING | EASE_OUT,
+		flags = ANIMATION_PARALLEL,
+	)
+
+	animate(
+		alpha = 0,
```
Comment (Mothblocks) [inline code/modules/balloon_alert/balloon_alert.dm:44]: wait this isnt what i meant at all

Comment (Mothblocks) [reply code/modules/balloon_alert/balloon_alert.dm:44]: oh well i like how it looks ill just make the code better

**Comment by YakumoChen** [comment]
I'd like to see this applied to mining in some way since the chat tends to clog badly with 1000x different "you smash the rock" messages

**Comment by Mothblocks** [comment]
>  actually does this remove text from the chat window or does it do both now

It does both for now, though there's no *need* to have both. For something like mining, I don't see any problem in replacing the chat message with it.

**Comment by Mothblocks** [comment]
Though to add on, while it's fine to add it to that, the ultimate goal here is for things that don't have any feedback other than chat.

Mining a rock has the feedback of a sound + the do_after prompt, so it doesn't particularly need this.

**Comment by RaveRadbury** [comment]
One potential point of confusion I see here is `information that is shown to everyone vs. information that is shown to you`

I assume that these transfer amount changes aren't shown to people next to me?

**Comment by Mothblocks** [comment]
They are only shown to you.

Also, I'm removing the chat message since it's now unnecessary and I want to keep the chat box uncluttered.

**Comment by RaveRadbury** [comment]
> They are only shown to you.

There's gotta be some convenient way we can indicate that, as this might become confusing as this gets used more and more

**Comment by PJB3005** [comment]
That'll be 5% of your revenue in patent licensing, thanks.

**Comment by Mothblocks** [comment]
> There's gotta be some convenient way we can indicate that, as this might become confusing as this gets used more and more

I don't think there is, but I also don't think it'll be much of a problem :shrug:

**Comment by zxaber** [comment]
Does mining even *need* a message about smashing rocks?

**Comment by Mothblocks** [comment]
Probably not, no.

**Comment by LemonInTheDark** [comment]
This looks really nice, I'm surprised we can pull it off in byond

**Comment by carshalash** [comment]
They're pretty slow.

**Comment by Mothblocks** [comment]
Slower than in the video, or just slow in general? If the former, I have no idea why--the client sides should be done 100% client-side. If the latter, I'd have to see it in game, since I thought I got a good sweet spot here.

**Pull request merged**